src/ctr_exit.c

```diff
@@ -156,9 +150,18 @@ void do_exit_command()
 	if (exit_pid) {
 		int ret, exit_status = 0;
 
-		do
-			ret = waitpid(exit_pid, &exit_status, 0);
-		while (ret < 0 && errno == EINTR);
+		/* even though we've already registered reap_children,
+		 * atexit() runs functions in reverse, so we need to
+		 * manually call here. Repeated calls will not cause trouble
+		 */
```

Comment:
I think this comment needs updating because we're not calling reap_children here